src/main/java/com/aws/greengrass/smbridge/clients/SMClient.java

```diff
@@ -5,129 +5,160 @@
 
-import javax.inject.Inject;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import javax.inject.Inject;
 
 public class SMClient {
     private static final Logger LOGGER = LogManager.getLogger(SMClient.class);
 
-    private StreamManagerClient smClient;
-    // The default stream receives all messages on reserved topics that have not been configured
-    // to go to another user-configured stream.
-    private MessageStreamDefinition defaultStream;
-    // TODO: Configurable MessageStreamDefinition for default settings of a specified but not-yet-created stream
+    private StreamManagerClient streamManagerClient;
+    private StreamDefinition streamDefinition;
+    private MessageStreamDefinition defaultStreamDefinition;
+
     /**
      * Ctr for SMClient.
      *
-     * @param topics             topics passed in by Nucleus
+     * @param  topics            topics passed in by Nucleus
+     * @param  streamDefinition  stream configs as parsed by SMBridge class
      * @throws SMClientException if unable to create SM Client
      */
     @Inject
-    public SMClient(Topics topics) throws SMClientException {
-        this(topics, null);
+    public SMClient(Topics topics, StreamDefinition streamDefinition) throws SMClientException {
+        this(topics, streamDefinition, null);
+        // TODO: Handle the case when serverUri is modified
         try {
-            // TODO: Discover non-default SM port
-            //  https://docs.aws.amazon.com/greengrass/v2/developerguide/use-stream-manager-in-custom-components.html
-            this.smClient = StreamManagerClientFactory.standard().build();
+            connectToStreamManagerWithCustomPort();
         } catch (StreamManagerException e) {
             throw new SMClientException("Unable to create a SM client", e);
         }
     }
-    protected SMClient(Topics topics, StreamManagerClient streamManagerClient){
-        this.smClient = streamManagerClient;
-        // TODO: Configurable default stream
-        this.defaultStream = new MessageStreamDefinition();
-        this.defaultStream.setName("mqttToStreamDefaultStreamName");
-        this.defaultStream.setTimeToLiveMillis(9223372036854L);
-        this.defaultStream.setStrategyOnFull(StrategyOnFull.RejectNewData);
+
+    @SuppressWarnings("PMD.UnusedFormalParameter") // topics may be needed later for extensibility with kernel interact
+    protected SMClient(Topics topics, StreamDefinition streamDefinition, StreamManagerClient streamManagerClient) {
+        this.streamManagerClient = streamManagerClient;
+        this.streamDefinition = streamDefinition;
     }
 
-    public void start() throws SMClientException {
+    @SuppressWarnings("PMD.CloseResource") // The socket is closed with a .disconnect() that pmd doesn't parse
+    void connectToStreamManagerWithCustomPort() throws StreamManagerException {
```

Comment:
Same comment. Isn't this component running as a plugin?

Reply:
Yes, will update

Reply:
Done!